vms/platformvm/warp/README.md

```diff
@@ -0,0 +1,121 @@
+# Avalanche Warp Messaging
+
+Avalanche Warp Messaging (AWM) provides a primitive for cross-subnet communication on the Avalanche Network.
+
+The Avalanche P-Chain provides an index of every Subnet's validator set on the Avalanche Network, including the BLS public key of each validator (as of the [Banff Upgrade](https://github.com/ava-labs/avalanchego/releases/v1.9.0)). AWM utilizes the weighted validator sets stored on the P-Chain to build a cross-subnet communication protocol between any two Subnets on the Avalanche Network.
+
+Any Virtual Machine (VM) on Avalanche can integrate Avalanche Warp Messaging to send and receive messages across Avalanche Subnets.
+
+## Background
+
+This README assumes familiarity with:
+
+- Avalanche P-Chain / [PlatformVM](../)
+- [ProposerVM](../../proposervm/README.md)
+- Basic familiarity with [BLS Multi-Signatures](https://crypto.stanford.edu/~dabo/pubs/papers/BLSmultisig.html)
+
+## BLS Multi-Signatures with Public-Key Aggregation
+
+Avalanche Warp Messaging utilizes BLS multi-signatures with public key aggregation in order to verify messages signed by another Subnet. When a validator joins a Subnet, the P-Chain records the validator's BLS public key and NodeID, as well as a proof of possession of the validator's BLS private key to defend against [rogue public-key attacks](https://crypto.stanford.edu/~dabo/pubs/papers/BLSmultisig.html#mjx-eqn-eqaggsame).
+
+AWM utilizes the validator set's weights and public keys to verify that an aggregate signature has sufficient weight signing the message from the source Subnet.
+
+BLS provides a way to aggregate signatures off chain into a single signature that can be efficiently verified on chain.
+
+## AWM Serialization
+
+Unsigned Message:
+```
++-----------------+----------+--------------------------+
+|      network_id :  uint32  |                 4 bytes  |
++-----------------+----------+--------------------------+
+| source_chain_id : [32]byte |                32 bytes  |
++-----------------+----------+--------------------------+
+|         payload :   []byte |       4 + size(payload)  |
++-----------------+----------+--------------------------+
+                             |  38 + size(payload) bytes|
+                             +--------------------------+
+```
+
+- `networkID` is the unique ID of an Avalanche Network (Mainnet/Testnet) and provides replay protection for BLS Signers across different Avalanche Networks
+- `sourceChainID` is the `blockchainID` of the blockchain sending a message. This ensures that each blockchain can only sign a message with its own `blockchainID`. Note: the `blockchainID` is the hash of the transaction that created the blockchain on the Avalanche P-Chain. It serves as the unique identifier for the blockchain across the Avalanche Network.
+- `payload` provides an arbitrary byte array containing the contents of the message. VMs define their own message types to include in the `payload`
+
+
+BitSetSignature:
+```
```

Comment:
Might be good to have a header on these tables to indicate what each column means (i.e field, type, size)

Reply:
Same comment as above